File: 프로그래머스/lv0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.swift
import Foundation

func solution(_ my_str:String, _ n:Int) -> [String] {
    var answer : [String] = []
    var temp = my_str
    while !temp.isEmpty {
        if (temp.count < n) {
            answer.append(temp)
            break
        }
        let index = temp.index(temp.startIndex, offsetBy: n)
        answer.append(String(temp[temp.startIndex..<index]))
        temp.removeSubrange(temp.startIndex..<index)
    }
    return answer
}

File: 프로그래머스/lv0/120913. 잘라서 배열로 저장하기/잘라서 배열로 저장하기.js
function solution(my_str, n) {
    const answer = [];
    while (my_str.length > 0) {
        const endIndex = my_str.length > n ? n : my_str.length;
        answer.push(my_str.slice(0, endIndex));
        my_str = my_str.slice(endIndex);
    }
    return answer;
}